frontend/packages/gitops-plugin/src/components/details/GitOpsResourcesSection.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {
+  Stack,
+  StackItem,
+  Split,
+  SplitItem,
+  Card,
+  CardBody,
+  Tooltip,
+} from '@patternfly/react-core';
+import { HeartBrokenIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import { ResourceIcon } from '@console/internal/components/utils';
+import { GitOpsEnvironmentService, GitOpsHealthResources } from '../utils/gitops-types';
+import './GitOpsResourcesSection.scss';
+
+interface GitOpsResourcesSectionProps {
+  services: GitOpsEnvironmentService[];
+  secrets: GitOpsHealthResources[];
+  deployments: GitOpsHealthResources[];
+  routes: GitOpsHealthResources[];
+  roleBindings: GitOpsHealthResources[];
+  clusterRoles: GitOpsHealthResources[];
+  clusterRoleBindings: GitOpsHealthResources[];
+}
+
+const getUnhealthyResources = () => (acc: string[], current: GitOpsHealthResources): string[] =>
+  current.health === 'Degraded' ||
+  current.health === 'Progressing' ||
+  current.health === 'Missing' ||
+  current.health === 'Unknown'
+    ? [...acc, current.health]
+    : acc;
+
+const getNonSyncedResources = () => (acc: string[], current: GitOpsHealthResources): string[] =>
+  current.status !== 'Synced' ? [...acc, current.status] : acc;
+
+const GitOpsResourcesSection: React.FC<GitOpsResourcesSectionProps> = ({
+  services,
+  secrets,
+  deployments,
+  routes,
+  roleBindings,
+  clusterRoles,
+  clusterRoleBindings,
+}) => {
+  const { t } = useTranslation();
+  const degradedServices: string[] = services ? services.reduce(getUnhealthyResources(), []) : [];
+  const degradedDeployments: string[] = deployments
+    ? deployments.reduce(getUnhealthyResources(), [])
+    : [];
+  const degradedSecrets: string[] = secrets ? secrets.reduce(getUnhealthyResources(), []) : [];
+  const degradedRoutes: string[] = routes ? routes.reduce(getNonSyncedResources(), []) : [];
+  const degradedRoleBindings: string[] = roleBindings
+    ? roleBindings.reduce(getNonSyncedResources(), [])
+    : [];
+  const degradedClusterRoles: string[] = clusterRoles
+    ? clusterRoles.reduce(getNonSyncedResources(), [])
+    : [];
+  const degradedClusterRoleBindings: string[] = clusterRoleBindings
+    ? clusterRoleBindings.reduce(getNonSyncedResources(), [])
+    : [];
+
+  const renderResource = (
+    resources: GitOpsHealthResources[] | GitOpsEnvironmentService[],
+    degradedResources: string[],
+  ) => {
+    return (
+      <Tooltip
+        content={t('gitops-plugin~{{x}} of {{total}} degraded', {
+          x: degradedResources.length.toString(),
+          total: resources ? resources.length.toString() : '0',
+        })}
+      >
+        <StackItem>
+          {degradedResources.length > 0 ? (
+            <>
+              {degradedResources.length}
+              <HeartBrokenIcon
+                color="#C9190B"
+                style={{ marginLeft: 'var(--pf-global--spacer--sm)' }}
+              />
+            </>
+          ) : (
+            <>&nbsp;</>
```

Comment:
Hmmm... If this happens you'll end up with a tooltip wrapping around nothing, right? Thus not really be able to be invoked.

Reply:
Added to issue [GITOPS-1280](https://issues.redhat.com/browse/GITOPS-1280)